python/ray/train/tests/test_minimal.py

```diff
@@ -53,27 +43,23 @@ def train_func():
         train.save_checkpoint(**checkpoint)
         return checkpoint[key]
 
-    checkpoint = {key: value}
-    test_callback = TestCallback()
-
-    trainer = Trainer(config, num_workers=num_workers)
-    trainer.start()
-    results = trainer.run(train_func, checkpoint=checkpoint, callbacks=[test_callback])
+    checkpoint = Checkpoint.from_dict(
+        {
+            # this would be set during checkpoint saving
+            "_current_checkpoint_id": 1,
+            key: value,
+        }
+    )
 
-    # Test results.
-    assert len(results) == num_workers
-    assert all(result == 1 for result in results)
+    trainer = DataParallelTrainer(
+        train_func,
+        backend_config=config,
+        resume_from_checkpoint=checkpoint,
+        scaling_config=dict(num_workers=num_workers),
+    )
+    results = trainer.fit()
 
-    # Test reporting and callbacks.
-    assert len(test_callback.result_list) == value
-    assert len(test_callback.result_list[0]) == num_workers
-    print(test_callback.result_list[0])
-    assert all(result[key] == value for result in test_callback.result_list[0])
-
-    # Test checkpointing.
-    assert trainer.latest_checkpoint[key] == value
-
-    trainer.shutdown()
+    assert results.checkpoint
```

Comment:
should be 
```suggestion
    assert results.checkpoint == checkpoint
```
?

Reply:
checkpoints cannot be compared directly, and the current checkpoint id will differ. I'll test for equality of `key value` though